src/helper_functions.hpp

```diff
@@ -1073,3 +1184,155 @@ matrix<Type> sim_pop(array<Type> NAA_devs, int recruit_model, vector<Type> mean_
   out.col(out.cols()-1) = SSB;
   return(out);
 }
+
+template <class Type>
+array<Type> pred_LAA(matrix<Type> mLAA_jan1, int n_yrs, int n_years_model, vector<Type> expSD,
+					  array<Type> GW_par, vector<Type> lengths, vector<Type> fracyr_vec, int growth_model, Type age_L1){
+
+  Type Lminp = min(lengths);
+  Type Lmaxp = max(lengths);
+  Type Fac1 = 0.0;
+  Type Fac2 = 0.0;
+  Type Ll1p = 0.0;
+  Type Llp = 0.0;
+  int n_ages = mLAA_jan1.cols();
+  int n_lengths = lengths.size();
+  array<Type> out(n_yrs, n_lengths, n_ages);
+  vector<Type> mLAA(n_ages);
+  Type Grate = 0.0;
+  Type b_len = 0.0;
+  Type last_linear = 0.0;
+  Type this_SD = 0.0;
+  Type Slope = 0.0;
+
+	for(int y = 0; y < n_yrs; y++) {
+		  mLAA.setZero();
+		  int y_1 = y + 1;
+		  int yuse = y; // used in fraction information
+		  if(y > (n_years_model - 1)) yuse = n_years_model -1; //some things only go up to n_years_model-1
+		  if(y == (n_yrs - 1)) y_1 = y;
+		  Type fracyr = fracyr_vec(yuse);
+		  for(int a = 0; a < n_ages; a++)
+		  {  
+			if(growth_model == 1) { // classic von Bertalanffy curve
+				b_len = (GW_par(y,a,2) - Lminp)/age_L1;
+				if((a + 1.0 + fracyr) <= age_L1) { // use linear growth
+					mLAA(a) = Lminp + b_len*(a+1.0+fracyr);  
+				} else { // other options
+					if((a+1.0) >= age_L1) { // only growth curve
+						if(a == (n_ages-1)) mLAA(a) = mLAA_jan1(y,a); // no growth for age plus group
+						else mLAA(a) = mLAA_jan1(y,a) + (mLAA_jan1(y,a) - GW_par(y,a,1))*(exp(-GW_par(y,a,0)*fracyr) - 1.0);
+					} else { // linear + growth curve mixed
+						last_linear = Lminp + b_len*age_L1; // last age (cont) with linear growth 
+						mLAA(a) = last_linear + (last_linear - GW_par(y,a,1))*(exp(-GW_par(y,a,0)*(a+1.0+fracyr-age_L1)) - 1.0); 
+					}
+				}
+			}
+			if(growth_model == 2) { // Richards growth curve	
+				b_len = (GW_par(y,a,2) - Lminp)/age_L1;			
+				if((a + 1.0 + fracyr) <= age_L1) { // use linear growth
+					mLAA(a) = Lminp + b_len*(a+1.0+fracyr);  
+				} else { // other options
+					if((a+1.0) >= age_L1) { // only growth curve
+						if(a == (n_ages-1)) mLAA(a) = mLAA_jan1(y,a); // no growth for age plus group
+						else mLAA(a) = pow(pow(mLAA_jan1(y,a),GW_par(y,a,3)) + (pow(mLAA_jan1(y,a),GW_par(y,a,3)) - pow(GW_par(y,a,1),GW_par(y,a,3)))*(exp(-GW_par(y,a,0)*fracyr) - 1.0),1/GW_par(y,a,3));
+					} else { // linear + growth curve mixed
+						last_linear = Lminp + b_len*age_L1; // last age (cont) with linear growth 
+						mLAA(a) = pow(pow(last_linear,GW_par(y,a,3)) + (pow(last_linear,GW_par(y,a,3)) - pow(GW_par(y,a,1),GW_par(y,a,3)))*(exp(-GW_par(y,a,0)*(a+1.0+fracyr-age_L1)) - 1.0),1/GW_par(y,a,3));
+					}
+				}
+			}
+			if(growth_model == 3){ // nonparametric approach
```

Comment:
The bias is very small for lengths (~5%), so a linear interpolation may be just fine. However, Crane's approach is required for weights since the linear interpolation would not work (~70% bias for age 1 and ~20% for age 2).